chainer/functions/normalization/decorrelated_batch_normalization.py

```diff
@@ -0,0 +1,299 @@
+        self.y_hat_pca = U.dot(x_hat)  # PCA whitening
+        y_hat = self.eigvectors.dot(self.y_hat_pca)  # ZCA whitening
+
+        y = y_hat.reshape((C, b * g,) + x.shape[2:]).transpose(
+            (1, 0) + spatial_axis)
+        if self.groups > 1:
+            y = y.reshape((-1, c) + x.shape[2:])
+
+        # Update running statistics
+        q = x.size // C
+        adjust = q / max(q - 1., 1.)  # unbiased estimation
+        self.running_mean *= self.decay
+        self.running_mean += (1 - self.decay) * mean
+        self.running_projection *= self.decay
+        projection = self.eigvectors.dot(U)
+        self.running_projection += (1 - self.decay) * adjust * projection
+
+        return y,
+
+    def backward(self, indexes, grad_outputs):
+        gy, = grad_outputs
+
+        f = DecorrelatedBatchNormalizationGrad(
+            self.groups, self.eigvals, self.eigvectors, self.y_hat_pca)
+        return f.apply((gy,))
+
+
+class DecorrelatedBatchNormalizationGrad(function_node.FunctionNode):
+
+    def __init__(self, groups, eigvals, eigvectors, y_hat_pca):
+        self.groups = groups
+        self.eigvals = eigvals
+        self.eigvectors = eigvectors
+        self.y_hat_pca = y_hat_pca
+
+    def forward(self, inputs):
+        self.retain_inputs(())
+        gy = inputs[0]
+        xp = backend.get_array_module(gy)
+        spatial_ndim = len(gy.shape[2:])
+        spatial_axis = tuple(range(2, 2 + spatial_ndim))
+        b, c = gy.shape[:2]
+        g = self.groups
+        C = c // g
+        m = b * g
+        for i in spatial_axis:
+            m *= gy.shape[i]
+
+        if g > 1:
+            gy = gy.reshape((b * g, C) + gy.shape[2:])
+        gy_hat = gy.transpose((1, 0) + spatial_axis).reshape(C, -1)
+
+        eigvectors = self.eigvectors
+        eigvals = self.eigvals
+        y_hat_pca = self.y_hat_pca
+        gy_hat_pca = eigvectors.T.dot(gy_hat)
+        f = gy_hat_pca.mean(axis=1)
+
+        K = eigvals[:, None] - eigvals[None, :]
+        valid = K != 0
+        K[valid] = 1 / K[valid]
+        xp.fill_diagonal(K, 0)
+
+        V = xp.diag(eigvals)
+        V_sqrt = xp.diag(eigvals ** 0.5)
+        V_invsqrt = xp.diag(eigvals ** -0.5)
+
+        F_c = gy_hat_pca.dot(y_hat_pca.T) / gy.dtype.type(m)
+        M = xp.diag(xp.diag(F_c))
+
+        mat = K.T * (V.dot(F_c.T) + V_sqrt.dot(F_c).dot(V_sqrt))
+        S = mat + mat.T
+        R = gy_hat_pca - f[:, None] + (S - M).T.dot(y_hat_pca)
+        gx_hat = R.T.dot(V_invsqrt).dot(eigvectors.T).T
+
+        gx = gx_hat.reshape((C, b * g,) + gy.shape[2:]).transpose(
+            (1, 0) + spatial_axis)
+        if g > 1:
+            gx = gx.reshape((-1, c, ) + gy.shape[2:])
+
+        self.retain_outputs(())
+        return gx,
+
+    def backward(self, inputs, grad_outputs):
+        # TODO(crcrpar): Implement this.
+        raise NotImplementedError("Double backward is not implemented for"
+                                  " decorrelated batch normalizatin.")
```

Comment:
Can you change to single quotes as per convention? There's also a typo `normalizatin`.